File: Backend/Models/Entities/Account.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models.Entities;

public class Account
{
    [Key]
    public int Id { get; set; }
    [Required]
    public string AccountName { get; set; } = null!;
    [Required]
    public string AccountNumber { get; set; } = null!;
    [Column(TypeName = "decimal(18,2)")]
    public decimal Balance { get; set; }
    public int UserId { get; set; }
    public bool IsMain { get; set; }
    public string Status { get; set; } = "Active";
    
    // Digital Card Details (populated for Main Account)
    public string? ExpiryDate { get; set; }
    public string? CVV { get; set; }

    public User? User { get; set; }

    public ICollection<Transaction>? SentTransactions { get; set; }
    public ICollection<Transaction>? ReceivedTransactions { get; set; }
}

File: mobile/mhlangafin/lib/models/account.dart
class Account {
  final int id;
  final String accountName;
  final String accountNumber;
  double balance;
  final bool isMain;
  final String status;
  final String? expiryDate;
  final String? cvv;

  Account({
    required this.id,
    required this.accountName,
    required this.accountNumber,
    required this.balance,
    required this.isMain,
    required this.status,
    this.expiryDate,
    this.cvv,
  });

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      id: json['id'],
      accountName: json['accountName'],
      accountNumber: json['accountNumber'],
      balance: (json['balance'] ?? 0.0).toDouble(),
      isMain: json['isMain'] ?? false,
      status: json['status'] ?? 'Active',
      expiryDate: json['expiryDate'],
      cvv: json['cvv'],
    );
  }
}
